server/initialize.js

```diff
@@ -83,51 +83,57 @@ export default function initialize(apolloConfig = {}, meteorApolloConfig = {}) {
   };
 }
 
-function getContextCreator(meteorApolloConfig, defaultContextResolver) {
-  return async function getContext({ req, connection }) {
+function getContextCreator(meteorApolloConfig, initialApolloConfig) {
+  const {
+    context: defaultContextResolver,
+    meteorAccounts,
+  } = initialApolloConfig;
+
+  const baseContext = { db };
+  return async function getContext({ req: request, connection }) {
+    // This function is called whenever a normal graphql request is being made,
+    // as well as when a client initiates a new subscription. However, when a
+    // client subscribes, the request headers are not being send along. The
+    // websocket only send those on the onConnect event. We store them on the
+    // `connection.context` together with the parsed cookies, so we can
+    // reconstruct a fake request object to be used by the context creator.
+    const req = connection ? connection.context.req : request;
+
+    let userContext = {};
+    if (meteorAccounts !== false && Package['accounts-base']) {
+      const loginToken =
+        req.headers['meteor-login-token'] ||
+        req.cookies['meteor-login-token'] ||
+        req.connectionParams && req.connectionParams[AUTH_TOKEN_KEY];
```

Comment:
I don't think this last fallback (`req.connectionParams && req.connectionParams[AUTH_TOKEN_KEY];`) is required any more, as the `onConnect` is passing trough the headers. But then again, connectionParams also seemed useless to me in the `onConnect`, so I decided to leave it for unforeseen reasons and backward compatibility.